File: 441-arranging-coins/arranging-coins.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        low=1
        high=n
        while(low<=high):
            mid = (low+high)//2
            if n< (mid*(mid+1)//2):
                high=mid-1
            elif n>=(mid*(mid+1)//2):
                low=mid+1
        return low-1

File: 441-arranging-coins/arranging-coins.cpp
class Solution {
public:
    int arrangeCoins(int n) {
        int left=1;
        int right=n;
        int ans=1;
        while(left<=right){
            int stair=left+(right-left)/2;
            long long needCoin=1ll*stair*(stair+1)/2;
            if(n>=needCoin){
                ans=stair;
                left=stair+1;
            }
            else{
                right=stair-1;
            }
        }
        return ans;
    }
};